### ISSUE TITLE
Strange timeout behaviour 

### ISSUE THREAD

**Issue by ivaylo321** [playwright-community/playwright-go#406]
I am looking at this trace from one of my test executions - The test failed on asserting url `p.Expect.Page(p.Page).ToHaveURL(regexp.MustCompile(expected))`

![Screenshot 2024-01-18 150428](https://github.com/playwright-community/playwright-go/assets/147637125/8d1c0b01-0889-4569-a00d-fab905ae2a8d)

I have configured the timeout to be 50000ms, but for some reason the assertion failed after less than 1 second - why could this happen, I am confused?

I received this log in my CI:

```
Page URL expected to be 'warning'
Actual value: <nil> 
Call log:
 with timeout 50000ms
waiting for locator(':root')
```

**Comment by canstand** [comment]
`Page.Goto` default timeout 30000ms,  and `ToHaveURL`  depends on `playwright.NewPlaywrightAssertions(timeout)`, default 5000ms.

**Comment by ivaylo321** [comment]
I did set the `playwright.NewPlaywrightAssertions` to 50000ms and I see that it is set correctly in the log.

From the debugging I did I think the problem is with the version of Chrome I use. I need to test with Chrome 83, so I have downloaded it separately and I am passing it when launching Chromium (with the `ExecutablePath` option).
I noticed that every time the test fails in such a strange way it is always with Chrome 83 and never with the Chrome version that comes with Playwright. So it appears that somehow the Chrome 83 gets stuck or even crashes (sometimes I get `Target crashed ` message in the logs).
From what I understand using older versions of Chrome is not officially supported and issues can be expected, correct? If that is the case I guess I just have to assume the tests with Chrome 83 will be flaky.
